modules/core/04-channel/keeper/keeper.go

```diff
@@ -384,6 +384,24 @@ func (k Keeper) IterateChannels(ctx sdk.Context, cb func(types.IdentifiedChannel
 	}
 }
 
+// GetChannelsWithPortPrefix returns all channels with the specified port prefix.
+func (k Keeper) GetChannelsWithPortPrefix(ctx sdk.Context, portPrefix string) []types.IdentifiedChannel {
```

Comment:
Could you add a basic test in 04-channel keeper for this function? It'd be nice to know prefixes and full portIDs are supported